utils/src/array/fixed_bytes.rs

```diff
@@ -0,0 +1,161 @@
+use crate::{hex, Array, SizedSerialize};
+use std::{
+    cmp::{Ord, PartialOrd},
+    fmt::{Debug, Display},
+    hash::Hash,
+    ops::Deref,
+};
+use thiserror::Error;
+
+/// Errors returned by `Bytes` functions.
+#[derive(Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("invalid length")]
+    InvalidLength,
+}
+
+/// An `Array` implementation for fixed-length byte arrays.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[repr(transparent)]
+pub struct FixedBytes<const N: usize>([u8; N]);
+
+impl<const N: usize> FixedBytes<N> {
```

Comment:
I think this obviates the need for the `Digest` type in SHA256 (?)
If so, can you remove it in this PR or make an issue?

Reply:
As noted above, decided to skip for now 👍